sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -101,34 +106,71 @@ public SentryTracer(
   @Override
   public void scheduleFinish() {
     synchronized (timerLock) {
-      cancelTimer();
+      cancelIdleTimer();
+      cancelDeadlineTimer();
```

Comment:
hm, if I understood this correctly we'll basically reschedule the deadline timer everytime we reschedule the idle timer - although our goal is to not exceed the 30s duration limit for automatic transactions ever.  

With this approach we're just rescheduling the deadline timeout and I don't think it ever will be fired if there are endlessly-appearing child spans for an idle tx.

We should just probably schedule it once in the SentryTracer constructor [here](https://github.com/getsentry/sentry-java/blob/549cbb4657b9dea1048c4882e79504e87d6acc4f/sentry/src/main/java/io/sentry/SentryTracer.java#L97) and never reschedule, guaranteeing the transaction duration will not exceed 30s. Does that make sense?